File: mod28/main.php
<?php

function countNumbersWithSumGreaterThan10(array $numbers): int {
    $count = 0;
    foreach ($numbers as $number) {
        $sum = 0;
        $temp = $number;
        while ($temp > 0) {
            $sum += $temp % 10;
            $temp = (int)($temp / 10);
        }
        if ($sum > 10) {
            $count++;
        }
    }
    return $count;
}

echo "Введите список чисел, разделенных пробелами: ";
$input = trim(fgets(STDIN));
$inputNumbers = explode(" ", $input);

$numbers = [];
foreach ($inputNumbers as $numStr) {
    $num = (int)$numStr;
    if (is_numeric($numStr)) {
        $numbers[] = $num;
    } else {
        echo "Некорректный ввод: $numStr. Это значение будет пропущено.\n";
    }

}

$result = countNumbersWithSumGreaterThan10($numbers);
echo "Количество чисел с суммой цифр > 10: $result\n";

?>

File: mod28/main.go
package main

import (
        "bufio"
        "fmt"
        "os"
        "strconv"
        "strings"
)

func countNumbersWithSumGreaterThan10(numbers []int) int {
        count := 0
        for _, number := range numbers {
                sum := 0
                temp := number
                for temp > 0 {
                        sum += temp % 10
                        temp /= 10
                }
                if sum > 10 {
                        count++
                }
        }
        return count
}

func main() {
        reader := bufio.NewReader(os.Stdin)
        fmt.Print("Введите список чисел, разделенных пробелами: ")
        input, _ := reader.ReadString('\n')
        input = strings.TrimSpace(input)
        inputNumbers := strings.Split(input, " ")

        numbers := []int{}
        for _, numStr := range inputNumbers {
                num, err := strconv.Atoi(numStr)
                if err != nil {
                        fmt.Printf("Некорректный ввод: %s. Это значение будет пропущено.\n", numStr)
                        continue
                }
                numbers = append(numbers, num)
        }

        result := countNumbersWithSumGreaterThan10(numbers)
        fmt.Printf("Количество чисел с суммой цифр > 10: %d\n", result)
}